### ISSUE TITLE
surface errors when OAuth token is not authorized

### ISSUE THREAD

**Issue by cellog** [desktop/desktop#10895]
### Describe the bug

For my work, I have an organization that is authenticated using SAML. Occasionally (but not always) github desktop will not respond when clicking on "fetch" or "push" or "pull origin". It will just act like I didn't click it. If I switch windows and then back, it has succeeded. Sometimes there are cryptic errors, and sometimes it just doesn't work at all. If I remove the repo and re-add it, suddenly everything works again

### Version & OS

2.5.6, Mac OS X 10.5.7

### Expected behavior

A box pops up with the notice "you must grant your OAuth token access to this organization"

### Actual behavior

crickets...

### Logs

repo name censored with `****` below

```
2020-10-21T16:24:55.082Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: **** - cellog (has token)
2020-10-21T16:24:55.755Z - warn: [ui] fetchRepository: an error occurred for '****'
Error: Resource protected by organization SAML enforcement. You must grant your OAuth token access to this organization.
    at Ne (/webpack:/app/src/lib/http.ts:170:11)
    at it.fetchRepository (/webpack:/app/src/lib/api.ts:567:16)
    at Object.repositoryWithRefreshedGitHubRepository (/webpack:/app/src/lib/stores/app-store.ts:3427:11)
    at Object.withAuthenticatingUser (/webpack:/app/src/lib/stores/app-store.ts:5156:7)
2020-10-21T16:24:55.755Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: **** - cellog (has token)
```

**Comment by cellog** [comment]
also - I have already authorized my ssh key (obviously) - how do I authorize the OAuth key? I don't see any options in Github Desktop, nor do I have any idea what it is talking about.

**Comment by steveward** [comment]
@cellog you will need to have GitHub Desktop approved in your organization. See [this article](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/approving-oauth-apps-for-your-organization) for more information. You may need to ask someone to do this depending on your permission levels. It's strange that it would be working intermittently -- if approval doesn't get things working let me know and we can dig into this further.

**Comment by cellog** [comment]
Thanks for the fast reply. Since I opened this, I uninstalled the desktop app, revoked its permissions, reinstalled and now it seems to be working. Could it be that the organization was setup after I installed gh desktop?

One other detail: it was showing the organization as "other" but now it shows the actual org name.

So SOMETHING has gotten the picture internal to the app. I will report back if it goes back to its old nasty tricks tomorrow. Also, out of curiosity, I have several other apps that asked me if I want to give or revoke permissions to specific orgs but gh desktop does not. Any reason for this?

Thanks!
